src/player.js

```diff
@@ -163,22 +163,29 @@ class Player {
      * Get a promise for a method.
      *
      * @param {string} name The API method to call.
-     * @param {Object} [args={}] Arguments to send via postMessage.
+     * @param {...(string|number|object|Array)} args Arguments to send via postMessage.
      * @return {Promise}
      */
-    callMethod(name, args = {}) {
+    callMethod(name, ...args) {
+        if (name === undefined || name === null) {
+            throw new TypeError('You must pass a method name.');
+        }
+
         return new Promise((resolve, reject) => {
-            // We are storing the resolve/reject handlers to call later, so we
-            // can’t return here.
-            // eslint-disable-next-line promise/always-return
-            return this.ready().then(() => {
-                storeCallback(this, name, {
-                    resolve,
-                    reject
-                });
+            return this.ready()
+                .then(() => {
+                    storeCallback(this, name, {
+                        resolve,
+                        reject
+                    });
+
+                    if (args.length > 1 || (args.length === 1 && Array.isArray(args[0]))) {
+                        return postMessage(this, name, args);
+                    }
+                    return postMessage(this, name, args[0]);
```

Comment:
```suggestion
                    return postMessage(this, name, args);
```
I'm trying to wrap my head around why this wouldn't work as is. When the message is received, wouldn't it still be able to handle a `value` that's an array but only one item? I may be missing something, but I think the `...args` change above may be enough.

Reply:
We do this now because we're now defining `callMethod` with a rest parameter. JavaScript automatically converts all of these arguments (`args`) into an array. 

If we always passed `args`, single values would be wrapped in an array unnecessarily. The player would receive ['red'] instead of just 'red', or `[['red', 'blue', 'green', 'white']]` instead of `['red', 'blue', 'green', 'white']`. Methods expecting single values would break if we don't unwrap single-value arrays.

Reply:
>  When the message is received, wouldn't it still be able to handle a value that's an array but only one item?

But yeah, to this point, you should be right. The receiver _should_ be able to handle this. I pushed up your suggestion. Thank you.